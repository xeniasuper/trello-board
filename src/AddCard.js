import React from "react"

class AddCard extends React.Component {
    render() {
        return (
            <div className={"add-card flex"}>
                <div className="plus-icon"></div>
                <p onClick={this.props.createCard}
                >
                    Добавить еще одну карточку
                </p>
            </div>
        )
    }
}

export {AddCard}