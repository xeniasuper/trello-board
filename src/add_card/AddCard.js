import React from "react"
import "./add_card.css"

class AddCard extends React.Component {
    render() {
        return (
            <div className={"add-card flex"}>
                <div className="plus-icon"></div>
                <p onClick={this.props.createCard}
                >
                    Добавить еще одну карточку
                </p>
            </div>
        )
    }
}

export {AddCard}