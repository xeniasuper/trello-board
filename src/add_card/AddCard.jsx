import React from 'react';
import './add_card.css';
import PropTypes from 'prop-types';

export default function AddCard(props) {
  const { createCard } = props;
  return (
    <div className="add-card flex">
      <div className="plus-icon" />
      <p
        id="add-one-more-card"
        onClick={createCard}
      >
        Добавить еще одну карточку
      </p>
    </div>
  );
}

AddCard.propTypes = {
  createCard: PropTypes.func.isRequired,
};
