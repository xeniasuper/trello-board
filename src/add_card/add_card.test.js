import React from 'react';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import AddCard from './AddCard';

Enzyme.configure({ adapter: new Adapter() });

describe('AddCard', () => {
  it('displays AddCard correctly', () => {
    const wrapper = Enzyme.shallow(<AddCard createCard={() => {}} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
