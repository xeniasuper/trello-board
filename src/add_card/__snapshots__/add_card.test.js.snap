// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCard displays AddCard correctly 1`] = `
<div
  className="add-card flex"
>
  <div
    className="plus-icon"
  />
  <p
    id="add-one-more-card"
    onClick={[Function]}
  >
    Добавить еще одну карточку
  </p>
</div>
`;
